src/sentry/utils/snuba.py

```diff
@@ -237,6 +237,8 @@ def zerofill(data, start, end, rollup, orderby):
         (end, start) = (start, end)
         rollup = rollup * -1
 
+    start = start + rollup
```

Comment:
does this just undo the `- rollup` at the end of line 232?

Reply:
Not exactly because we swap start/end depending on ordering

Reply:
hmm, in that case I guess I don't quite understand what this does. In the normal (unswapped) case it moves `start` closer to `end`, decreasing your total number of buckets. And in the swapped case, it moves `start` away from `end`, increasing the number of buckets?

Reply:
Ah yeah that swap is wrong since it changes the boundaries - changed the logic to just reverse the list at the end if necessary.